modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -524,6 +524,64 @@ std::shared_ptr<DataFrame> combineDataFrames(std::vector<std::shared_ptr<DataFra
     return newDataFrame;
 }
 
+#include <warn/push>
+#include <warn/ignore/conversion>
+std::vector<std::uint32_t> filteredRows(const Column& col, dataframefilters::Filters filters) {
+
+    if (col.getColumnType() == ColumnType::Categorical) {
+        const auto& catCol = dynamic_cast<const CategoricalColumn&>(col);
+        std::vector<std::uint32_t> rows;
+        for (auto&& [row, value] : util::enumerate<std::uint32_t>(catCol.getValues())) {
+            auto test = util::overloaded{
+                [v = value](const std::function<bool(std::string_view)>& func) { return func(v); },
+                [](const auto&) { return false; }};
+            if (std::any_of(filters.include.begin(), filters.include.end(),
+                            [&](const auto& f) { return std::visit(test, f.filter); })) {
+                rows.push_back(row);
+            }
+        }
+        return rows;
+    } else {
+        return col.getBuffer()
+            ->getRepresentation<BufferRAM>()
+            ->dispatch<std::vector<std::uint32_t>, dispatching::filter::Scalars>(
+                [filters](auto typedBuf) {
+                    using ValueType = util::PrecisionValueType<decltype(typedBuf)>;
+                    std::vector<std::uint32_t> rows;
+                    for (auto&& [row, value] :
+                         util::enumerate<std::uint32_t>(typedBuf->getDataContainer())) {
+                        auto test = util::overloaded{
+                            [v = value](const std::function<bool(int)>& func) {
+                                if constexpr (std::is_integral_v<ValueType>) {
```

Comment:
Could this test not be outside the loop? should the non matching case not return true? skip the const auto& case as above.